File: JavaCoding/Array/TwoDArray/transposeMatrix.java
package JavaCoding.Array.TwoDArray;

import java.util.Arrays;

public class transposeMatrix {

    public static void main(String[] args) {
        

       int [][] matrix = {{1,2,3},
                          {4,5,6}};
       int [][] result = transpose(matrix);

      int row = result.length;
      int coloum = result[0].length;
      
        for(int i =0 ;i<row;i++)
      {
        for(int j =0 ;j<coloum;j++){

           System.out.print(result[i][j] + " ");

        }
        
           System.out.println();
      }
    }


    public static int [][] transpose(int [][] arr){
      int row = arr.length;
      int coloum = arr[0].length;

      int [][] transpose =  new int[coloum][row];

     
      for(int i =0 ;i<row;i++)
      {
        for(int j =0 ;j<coloum;j++){

            transpose[j][i]=arr[i][j];

        }
      }

        return transpose;


    }
    
    
}

File: Javascript/Array/twoDArray/transposeMatrix.js

let matrix = [[1,2,3],[4,5,6]];

let result = transpose(matrix);

     let row = result.length;
let coloum = result[0].length;

for (let i = 0; i < row; i++) {
    let line = "";
    for (let j = 0; j < coloum; j++) {
        line += result[i][j] + " ";
    }
    console.log(line.trim());

}
function transpose(arr){

     let rows = arr.length;
     let coloums = arr[0].length;


     // Proper 2D array initialization
    let transpose = new Array(coloums).fill(0).map(() => new Array(rows));

     for(let i = 0 ; i<rows;i++){


        for(let j=0;j<coloums;j++){


            transpose[j][i] = arr[i][j];
        }
     }


     return transpose;

}